base/compiler/utilities.jl

```diff
@@ -349,6 +369,51 @@ end
 @inline slot_id(s) = isa(s, SlotNumber) ? (s::SlotNumber).id :
     isa(s, Argument) ? (s::Argument).n : (s::TypedSlot).id
 
+######################
+# IncrementalCompact #
+######################
+
+# specifically meant to be used with body1 = compact.result and body2 = compact.new_new_nodes, with nvals == length(compact.used_ssas)
+function find_ssavalue_uses1(body1::Vector{Any}, body2::Vector{Any}, nvals::Int)
+    uses = zeros(Int, nvals)
+
+    function increment_uses(ssa::SSAValue)
+        uses[ssa.id] += 1
+    end
+
+    for line in 1:(length(body1) + length(body2))
+        # index into the right body
+        if line <= length(body1)
+            isassigned(body1, line) || continue
+            e = body1[line]
+        else
+            nline = line - length(body1)
+            isassigned(body2, nline) || continue
+            e = body2[nline]
+        end
+
+        # only look at relevant (in terms of uses) fields
```

Comment:
Should be able to drop this if/else - userefs should handle it.

Reply:
👍